### DISCUSSION TITLE
[Feature Request] Controlled disclosure

### DISCUSSION THREAD

**Discussion by maximelebastard** [tailwindlabs/headlessui#2692] [Ideas]
**Use case**

I'm using Disclosure for a navigation. For user comfort I'm storing which menus are opened and which are not.

**Usage example**

I'd like to do that
```
const [isOpened, saveOpened] = usePersistedState(false); // this is an example state that is persisted
<Disclosure open={isOpened} onOpen{saveOpened}>
  <Disclosure.Button>Open</Disclosure.Button>
</Disclosure>
```

**Current workaround**
I could listen to all my Disclosure.Button to store changes and use defaultOpen to pass the initial state. But it seems quite complicated and bug prone for just a disclosure state. It kind of reduces the value of using headless ui

**Comment by olivierpascal** [comment]
Yes, please!

**Comment by tacomanator** [comment]
I too find myself wanting to be able to preserve the open/close state of a disclosure on navigating away and back again.

**Comment by spacecat** [comment]
Me too. How's it going with the new APIs for the entire component catalogue?

**Comment by runoncedev** [comment]
There's still no way to do this, right? Not even on v2
